File: lessons/day_05/java/src/main/java/com/example/DemoGraphQL/input/InputEngineerCreate.java
package com.example.DemoGraphQL.input;

import com.example.DemoGraphQL.model.EyeColor;
import com.example.DemoGraphQL.model.Grade;
import com.example.DemoGraphQL.model.Role;

import java.util.List;

public record InputEngineerCreate (
    String name,
    String surname,
    String email,
    Integer age,
    EyeColor eyeColor,
    Long favSkill,
    List<Long> friends,
    List<Long> skills,
    Role role,
    Grade grade
){}

File: lessons/day_06/netcore/src/GraphQLSample/Models/Input/InputEngineerCreate.cs
using GraphQLNetCore.Models.Enums;
using System.Collections.Generic;

namespace GraphQLNetCore.Models.Input
{
   public class InputEngineerCreate
   {
      public string Name { get; set; }
      public string Surname { get; set; }
      public string Email { get; set; }
      public int Age { get; set; }
      public EyeColor? EyeColor { get; set; }
      public List<int> Friends { get; set; }
      public List<int> Skills { get; set; }
      public int? FavSkill { get; set; }
      public int EmployeeId { get; set; }
      public Role Role { get; set; }
      public Grade Grade { get; set; }

      internal Engineer ToPerson()
      {
         return new Engineer() {
            Age = Age,
            Email = Email,
            EyeColor = EyeColor,
            FavSkillId = FavSkill,
            Name = Name,
            Surname = Surname,
            EmployeeId = EmployeeId,
            Role = Role,
            Grade = Grade,
         };
      }
   }
}
